lib/smart_proxy_dynflow_core/log.rb

```diff
@@ -1,61 +1,106 @@
-require 'logger'
+require 'logging'
 
 module SmartProxyDynflowCore
-  class Log < ::Logger
-    alias_method :write, :debug
+  class Log
+    BASE_LOG_SIZE = 1024 * 1024 # 1 MiB
+    LOGGER_NAME = 'dynflow-core'
 
-    class << self
-      def instance
-        if @logger.nil?
-          @logger = self.new log_file
-          @logger.level = log_level
-        end
-        @logger
-      end
-
-      def instance=(logger)
-        @logger = logger
-      end
+    begin
+      require 'syslog/logger'
+      @syslog_available = true
+    rescue LoadError
+      @syslog_available = false
+    end
 
+    class << self
       def reload!
+        Logging.logger[LOGGER_NAME].appenders.each(&:close)
+        Logging.logger[LOGGER_NAME].clear_appenders
         @logger = nil
         instance
       end
 
-      def log_level
-        if Settings.instance.loaded && Settings.instance.log_level
-          ::Logger.const_get(Settings.instance.log_level.upcase)
+      def reopen
+        return if @logger.nil?
+        @logger.appenders.each(&:reopen)
+      end
+
+      def instance
+        return @logger if @logger
+        layout = Logging::Layouts.pattern(pattern: setting(:file_logging_pattern, '%d %.8X{request} [%.1l] %m') + "\n")
+        notime_layout = Logging::Layouts.pattern(pattern: setting(:system_logging_pattern, '%.8X{request} [%.1l] %m') + "\n")
+        log_file = setting(:log_file, 'STDOUT')
```

Comment:
The defaults are being set here https://github.com/theforeman/smart_proxy_dynflow/blob/37e9a46f13f3d1b572a41ea471713fa13b631dbf/lib/smart_proxy_dynflow_core/settings.rb#L38